### PR TITLE
[FIX] MDS: No optimization when subset data

### PR DESCRIPTION
**Pull request by jerneju**
##### Issue
~~1. MDS cannot handle sparse data.~~ That will be fixed in #2690 .
2. Optimization runs when subset data is sent to the widget. This is not necessary.
3. Plot resets when _Show similar pairs_ slider is moved.
4. Similar pairs are not removed when _Show similar pairs_ slider is set to 0. And they should be removed.

##### Includes
- [X] Code changes
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by jerneju** [comment]
https://sentry.io/biolab/orange3/issues/367233902/

**Comment by janezd** [comment]
Add tests for sparse data.

**Comment by ajdapretnar** [comment]
This now works!
But sending Data Subset to MDS does not do anything. I mean it does not hollow out the points as it should... Am I missing something?

File: Orange/widgets/unsupervised/owmds.py [hunk 1]
```diff
@@ -651,18 +652,21 @@ def _setup_plot(self, new=False):
 
         data = self.data
 
-        primitive_metas = tuple(a for a in data.domain.metas if a.is_primitive())
-        keys = [k for k, a in enumerate(data.domain.metas) if a.is_primitive()]
-        data_metas = data.metas[:, keys]
-
-        attributes = self.data.domain.attributes + (self.variable_x, self.variable_y) + \
-                     primitive_metas
+        if self.data.is_sparse():
```
Comment (nikicc) [inline Orange/widgets/unsupervised/owmds.py]: Why do we need a different logic for sparse here?

**Comment by nikicc** [comment]
Also, on hover doesn't show all data for sparse. Is there a special reason for this?

File: Orange/widgets/unsupervised/owmds.py [hunk 2]
```diff
@@ -271,6 +271,8 @@ def update_regression_line(self):
 
     def init_attr_values(self):
         domain = self.data and len(self.data) and self.data.domain or None
+        if domain is not None and self.data.is_sparse():
+            domain = Domain(attributes=[], class_vars=domain.class_vars)
```
[the remark this answers is not in the document]
Comment (nikicc) [reply Orange/widgets/unsupervised/owmds.py]: #2690 should probably solve this issue.

File: Orange/widgets/unsupervised/owmds.py (hunk 1 above)
Comment (jerneju) [reply Orange/widgets/unsupervised/owmds.py]: #2693

**Pull request merged**